File: DynamicTypes/benchmarks/fib-recursive.py
#!/usr/bin/env python3
"""Recursive Fibonacci sequence."""

import time

def rfib(n):
    if n < 1:
        return 0
    if n == 1:
        return 1
    return rfib(n-1) + rfib(n-2)

def run_benchmark(n=40):
    t0 = time.time()
    x = rfib(n)
    t1 = time.time()
    elapsed = t1 - t0
    print(f"rfib({n}) = {x}, time: {elapsed:.3f} seconds")
    results[f"rfib({n})"] = round(elapsed, 3)

if __name__ == "__main__":
    results = {}
    run_benchmark()

File: DynamicTypes/benchmarks/fib-recursive.lua
-- Recursive Fibonacci sequence.

function rfib(n)
    if n < 1 then return 0 end
    if n == 1 then return 1 end
    return rfib(n-1) + rfib(n-2)
end

function runBenchmark(n)
    n = n or 40
    local t0 = os.clock()
    local x = rfib(n)
    local t1 = os.clock()
    local elapsed = t1 - t0
    print(string.format("rfib(%d) = %d, time: %.3f seconds", n, x, elapsed))
    results["rfib(" .. n .. ")"] = math.floor(elapsed * 1000 + 0.5) / 1000  -- round to 3 decimal places
end

-- Main execution (equivalent to "if locals == globals")
if not package.loaded[...] then
    results = {}
    runBenchmark()
end